File: Documentation/API/Utilities/CalculateAccuracy.cpp
#include <iostream>

#include "RhythmGameUtilities/Utilities.hpp"

using namespace RhythmGameUtilities;

auto main() -> int
{
    const int seconds = 2;
    const int resolution = 192;
    const int positionDelta = 50;

    std::vector<Tempo> tempoChanges = {{0, 120000}};

    auto note = new Note{1, 750};
    auto currentPosition =
        ConvertSecondsToTicks(seconds, resolution, tempoChanges);

    auto accuracy =
        CalculateAccuracy(note->Position, currentPosition, positionDelta);

    std::cout << ToString(accuracy) << std::endl; // Good

    return 0;
}

File: Documentation/API/Utilities/CalculateAccuracy.js
import RhythmGameUtilities from '@rhythm-game-utilities/core';

const seconds = 2;
const resolution = 192;
const positionDelta = 50;

const tempoChanges = [{ position: 0, bpm: 120000 }];

const currentPosition = RhythmGameUtilities.ConvertSecondsToTicks(seconds, resolution, tempoChanges);

const accuracy = RhythmGameUtilities.CalculateAccuracy(750, currentPosition, positionDelta);

console.log(accuracy); // Good
